jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -5,23 +5,363 @@
 /** Status React component used to display Juju status. */
 class Status extends React.Component {
 
+  /**
+    Render the component.
+  */
   render() {
     return (
       <juju.components.Panel
         instanceName="status-view"
         visible={true}>
         <div className="status-view__content">
-          Status
+          {this.renderStatus()}
         </div>
       </juju.components.Panel>
     );
   }
 
-};
+  /**
+    Render the current model status.
+  */
+  renderStatus() {
```

Comment:
@hatched would say these methods don't render the element, but rather generate the objects that are used to render them so instead of `renderStatus` they could be `_generateStatus`, but up to you if you want to change them or not.